### PR TITLE
HZC-2739 Add tutorial for the Spring Boot application

### PR DESCRIPTION
**Pull request by JakeSCahill**

### CODE REVIEW

File: spring-sample/docs/modules/ROOT/pages/spring-boot-client.adoc [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+:page-serverless: true
+:url-spring-boot: https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#getting-started
+:url-spring-hazelcast: https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#io.hazelcast
+:github-directory: https://github.com/hazelcast/hazelcast-cloud-code-samples/tree/master/spring-sample 
+
+{description}
+
+== Context
+
+Spring Boot helps you to create stand-alone, production-grade Spring-based applications. If you're unfamiliar with Spring or Spring Boot, see the link:{url-spring-boot}[Spring Boot documentation].
```
Comment (angelasimms) [inline spring-sample/docs/modules/ROOT/pages/spring-boot-client.adoc:17]: Maybe too many compound adjectives? Would it be clearer to say this, or something similar.

....stand-alone Spring-based applications that are production grade.

File: spring-sample/docs/modules/ROOT/pages/spring-boot-client.adoc [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+== Before you Begin
+
+You need the following:
+
+- A xref:cloud:ROOT:create-serverless-cluster.adoc[Serverless cluster].
+- link:https://git-scm.com/book/en/v2/Getting-Started-Installing-Git[Git]
+- link:https://maven.apache.org/install.html[Maven]
+- JDK 8 installed and set up as the `JAVA_HOME` environment variable.
```
Comment (angelasimms) [inline spring-sample/docs/modules/ROOT/pages/spring-boot-client.adoc:26]: Not sure of this wording.

JDK 8 installed and set up as the `JAVA_HOME` environment variable.

You set up a the environment variable for the JDK??

File: spring-sample/docs/modules/ROOT/pages/spring-boot-client.adoc [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+- `trustStorePassword` (same as `keyStorePassword`)
+
+. Download the keystore and truststore files from the {hazelcast-cloud} console and add them to the `spring-sample/src/main/resources` directory. You can find these files in the {hazelcast-cloud} console under *Connect Client* > *Advanced Setup*.
+
+The sample code for this tutorial is in the link:{github-directory}[`hazelcast-cloud-code-samples/spring-sample`] directory.
+
+.`pom.xml`
+[%collapsible]
+====
+If the Hazelcast client is on the classpath and a suitable configuration is found, Spring Boot auto-configures a `HazelcastInstance` that you can inject into your application.
```
Comment (angelasimms) [inline spring-sample/docs/modules/ROOT/pages/spring-boot-client.adoc:74]:
```suggestion
If the Hazelcast client is on the classpath and a suitable configuration is found, Spring Boot automatically configures a `HazelcastInstance` that you can inject into your application.
```

File: spring-sample/docs/modules/ROOT/pages/spring-boot-client.adoc [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+include::spring-sample:example$pom.xml[tag=client]
+----
+
+link:{github-directory}/pom.xml[View source]
+====
+
+.`HzViridianDemoApplication.java`
+[%collapsible]
+====
+Spring Boot first attempts to create a Hazelcast client by checking the presence of a `com.hazelcast.client.config.ClientConfig` bean. This bean is configured, using the information in the `application.properties` file.
```
Comment (angelasimms) [inline spring-sample/docs/modules/ROOT/pages/spring-boot-client.adoc:87]:
```suggestion
Spring Boot first attempts to create a Hazelcast client by checking the presence of a `com.hazelcast.client.config.ClientConfig` bean. This bean is configured using the information in the `application.properties` file.
```

File: spring-sample/docs/modules/ROOT/pages/spring-boot-client.adoc [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+include::spring-sample:example$src/main/java/sample/com/hazelcast/demo/viridian/HzViridianDemoApplication.java[tag=class]
+----
+
+link:{github-directory}/src/main/java/sample/com/hazelcast/demo/viridian/HzViridianDemoApplication.java[View source]
+====
+
+.`MapService.java`
+[%collapsible]
+====
+When the `com.hazelcast.client.config.ClientConfig` bean is present, a `HazelcastInstance` is injected into the application classes. This instance is a Hazelcast client with an established connection to the Hazelcast Viridian cluster.
```
Comment (angelasimms) [inline spring-sample/docs/modules/ROOT/pages/spring-boot-client.adoc:100]:
```suggestion
When the `com.hazelcast.client.config.ClientConfig` bean is present, a `HazelcastInstance` is injected into the application classes. This instance is a Hazelcast client with an established connection to the {hazelcast-cloud} cluster.
```

File: spring-sample/docs/modules/ROOT/pages/spring-boot-client.adoc [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+[source,java]
+----
```
Comment (angelasimms) [inline spring-sample/docs/modules/ROOT/pages/spring-boot-client.adoc:103]: 404 when I try to view MapService.java

File: spring-sample/docs/modules/ROOT/pages/spring-boot-client.adoc (hunk 1 above)
Comment (JakeSCahill) [reply spring-sample/docs/modules/ROOT/pages/spring-boot-client.adoc:17]: I agree. I actually took this from the Spring Boot docs. We can reword it.

File: spring-sample/docs/modules/ROOT/pages/spring-boot-client.adoc (hunk 6 above)
Comment (angelasimms) [reply spring-sample/docs/modules/ROOT/pages/spring-boot-client.adoc:108]: I can see that the link is pointing to the master branch and the existing SomeService.java will be overwritten by this file.

**Review by angelasimms — APPROVED** [review]

**Pull request merged**